app/src/pages/inside/testCaseLibraryPage/importTestCaseModal/importTestCaseModal.tsx

```diff
@@ -51,35 +55,74 @@ const toMB = (bytes: number) => +(bytes / (1024 * 1024)).toFixed(2);
 
 export const ImportTestCaseModal = ({
   handleSubmit,
+  change,
   data,
 }: InjectedFormProps<ImportTestCaseFormValues, ImportModalData> & ImportModalData) => {
   const { formatMessage } = useIntl();
   const [file, setFile] = useState<File | null>(null);
   const [folderIdFromUrl] = useState<number | undefined>(() =>
     extractFolderIdFromHash(window.location.hash),
   );
+  const [target, setTarget] = useState<'root' | 'existing'>(
+    folderIdFromUrl != null ? 'existing' : 'root',
+  );
+  const [existingFolderId, setExistingFolderId] = useState<number | null>(null);
+  const folders = useSelector(foldersSelector);
   const dispatch = useDispatch();
   const selectedFolderName = data?.folderName ?? '';
+  const hasFolderIdFromUrl = folderIdFromUrl != null;
 
   const { isImportingTestCases, importTestCases } = useImportTestCase();
 
   const handleCancel = () => {
     dispatch(hideModalAction());
   };
 
+  const setTargetAndForm = (next: 'root' | 'existing') => {
```

Comment:
Let's create type for `'root' | 'existing'` union and use here and in `useState` above